### PR TITLE
Fix a few Closure and JSDoc issues.

### PR DESCRIPTION
**Pull request by XhmikosR**

### CODE REVIEW

File: lighthouse-core/gather/driver.js [hunk 1]
```diff
@@ -255,6 +255,7 @@ class Driver {
    * If our main document URL redirects, we will update options.url accordingly
    * As such, options.url will always represent the post-redirected URL.
    * options.initialUrl is the pre-redirect URL that things started with
+   * @param {!string} opts
```
Comment (ebidel) [inline lighthouse-core/gather/driver.js]: !Object

File: lighthouse-core/lib/log.js [hunk 2]
```diff
@@ -38,6 +38,8 @@ class Emitter extends EventEmitter {
   /**
    * Fires off all status updates. Listen with
    * `require('lib/log').events.addListener('status', callback)`
+   * @param {string} title
+
```
Comment (ebidel) [inline lighthouse-core/lib/log.js]: remove extras newline and add `args` param

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js [hunk 3]
```diff
@@ -342,7 +342,7 @@ class LighthouseViewerReport {
 
   /**
    * Downloads a file (blob) using a[download].
-   * @param {Blob|File} The file to save.
+   * @param {string} blob The file to save.
```
Comment (ebidel) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: this should stay `Blob|File`

**Comment by XhmikosR** [comment]
Yup,  I'll address your comments later.

File: lighthouse-core/lib/log.js (hunk 2 above)
Comment (XhmikosR) [inline lighthouse-core/lib/log.js]: What type is `args`? `Object`?

File: lighthouse-core/lib/log.js [hunk 4]
```diff
@@ -38,6 +38,8 @@ class Emitter extends EventEmitter {
   /**
    * Fires off all status updates. Listen with
    * `require('lib/log').events.addListener('status', callback)`
+   * @param {string} title
+   * @param {function(...)} args
```
Comment (ebidel) [inline lighthouse-core/lib/log.js]: From callsite usage, it look like it's an optional `arguments` array.

Comment (ebidel) [inline lighthouse-core/lib/log.js]: https://nodejs.org/api/events.html#events_emitter_emit_eventname_args

File: lighthouse-core/lib/log.js [hunk 5]
```diff
@@ -48,6 +50,7 @@ class Emitter extends EventEmitter {
   /**
    * Fires off all warnings. Listen with
    * `require('lib/log').events.addListener('warning', callback)`
+   * @param {function(...)} args
```
Comment (ebidel) [inline lighthouse-core/lib/log.js]: non-optional `arguments` array

File: lighthouse-extension/app/src/popup.js [hunk 6]
```diff
@@ -121,6 +121,7 @@ document.addEventListener('DOMContentLoaded', _ => {
   /**
    * Generates a document fragment containing a list of checkboxes and labels
    * for the aggregation categories.
+   * @param {!Object<null>} list
```
Comment (ebidel) [inline lighthouse-extension/app/src/popup.js]: @brendankenny is `Object<null>` a thing? I only found https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#type-application

File: lighthouse-viewer/app/src/lighthouse-report-viewer.js [hunk 7]
```diff
@@ -267,6 +268,7 @@ class LighthouseViewerReport {
 
   /**
    * Enables pasting a JSON report on the page.
+   * @param {string} e Event name.
```
Comment (ebidel) [inline lighthouse-viewer/app/src/lighthouse-report-viewer.js]: you can remove "Event name". It's just an event and obvious. Remove throughout.

**Comment by XhmikosR** [comment]
@ebidel: I reverted the changes. I don't think it's worth me spending more time on this especially since closure doesn't run in CI.

So feel free to push here for further improvements, or close it, it's your call.

I was just trying to solve the easy Closure issues.

**Comment by ebidel** [comment]
Understood. Appreciate all your help so far. We can update the rest.

**Pull request merged**